tests/static/form/mismatch-tree-order.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<h:html xmlns:h="http://www.w3.org/1999/xhtml" xmlns:orx="http://openrosa.org/jr/xforms" xmlns="http://www.w3.org/2002/xforms" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:jr="http://openrosa.org/javarosa" xmlns:vellum="http://commcarehq.org/xforms/vellum">
+    <h:head>
+        <h:title>Untitled Form</h:title>
+        <model>
+            <instance>
+                <data xmlns:jrm="http://dev.commcarehq.org/jr/xforms" xmlns="http://openrosa.org/formdesigner/606D428D-4D37-43A5-81E2-EB793CE4E441" uiVersion="1" version="1" name="Untitled Form">
+                    <question1 />
+                    <question5 />
+                    <question2>
+                        <question3 />
+                    </question2>
+                    <question6 />
+                    <question4 />
+                </data>
+            </instance>
+            <bind nodeset="/data/question1" type="xsd:string" />
+            <bind nodeset="/data/question5" />
+            <bind nodeset="/data/question2" />
+            <bind nodeset="/data/question2/question3" type="xsd:string" />
+            <bind nodeset="/data/question6" />
+            <bind nodeset="/data/question4" type="xsd:string" />
+            <itext>
+                <translation lang="en" default="">
+                    <text id="question1-label">
+                        <value>question1</value>
+                    </text>
+                    <text id="question2-label">
+                        <value>question2</value>
+                    </text>
+                    <text id="question2/question3-label">
+                        <value>question3</value>
+                    </text>
+                    <text id="question2/question4-label">
+                        <value>question4</value>
+                    </text>
+                </translation>
+            </itext>
+        </model>
+    </h:head>
+    <h:body>
+        <input ref="/data/question1">
+            <label ref="jr:itext('question1-label')" />
+        </input>
+        <group ref="/data/question2">
+            <label ref="jr:itext('question2-label')" />
+            <input ref="/data/question2/question3">
+                <label ref="jr:itext('question2/question3-label')" />
+            </input>
+        </group>
+        <input ref="/data/question4">
+            <label ref="jr:itext('question2/question4-label')" />
+        </input>
```

Comment:
The data tree order here is 1, 5, 2, 6, 4 with 5 and 6 being data-only, and control tree order is 1, 2, 4. Those are not mismatched trees, correct? I think you want to change this so the control tree order is 1, 4, 2.

The old test did this:
```js
form.dataTree = makeTree("data", ["a", "x1", "b", "x2", "c"]); // a b c
form.controlTree = makeTree("control", ["a", "c", "b"]);       // a c b
```